feat: chat timestamp

agazso opened the pull request:
Closes #32 

There were several simplifications compared to the original design: 
- [ ] In the design the timestamps are on the same line if the message is only one line
- [ ] The badges indicating the days are floating in the design, I just implemented them as markers between the chat bubbles.

![Screenshot from 2023-09-21 18-18-59](https://github.com/logos-innovation-lab/waku-objects-playground/assets/230163/94c6d857-50cd-4d0b-9df5-a2dda0d0547e)

vojtechsimetka reviewed (approved):
So this is definitely an improvement, but it does not play nice with object. We can tackle that separately, just FYI

agazso commented:
> So this is definitely an improvement, but it does not play nice with object. We can tackle that separately, just FYI

What is the issue? Maybe I can address it in this PR.

vojtechsimetka commented:
Yes to few things:

1. Timestamps don't exist on waku object messages
2. In the following screenshot, I've sent the waku object today and then the text. However, the `Today` only appeared after the waku object and before the normal text. There is around 1 min difference between these two.
![Screenshot 2023-09-22 at 18 01 32](https://github.com/logos-innovation-lab/waku-objects-playground/assets/7974813/28d0e9c8-0a3a-4927-89d3-dcd976157623)

3. If I scroll all the way to the top, it shows just time and I have no idea when the message was sent (it was sent 2 days ago). So before the first message in the chat we should always have the date.
![Screenshot 2023-09-22 at 18 01 38](https://github.com/logos-innovation-lab/waku-objects-playground/assets/7974813/2def101c-06cc-4291-a042-3f5e8764c5ea)

agazso commented:
>     1. Timestamps don't exist on waku object messages

This is an existing issue and I didn't necessarily want to fix in this PR: https://github.com/logos-innovation-lab/waku-objects-playground/issues/347

>     2. In the following screenshot, I've sent the waku object today and then the text. However, the `Today` only appeared after the waku object and before the normal text. There is around 1 min difference between these two.

This is indeed an issue, and I will fix it.

>     3. If I scroll all the way to the top, it shows just time and I have no idea when the message was sent (it was sent 2 days ago). So before the first message in the chat we should always have the date.

This is a new feature request and I think it's a good idea and seems easy to add, so I will add it too.

agazso commented:
I added a fix that addresses points 2 and 3. 

As for the first one, currently all the objects render their own `ChatMessage` components and that's where the timestamp needs to be passed from the message (also the name and avatar in #347). It would be better if the host app would render the `ChatMessage` frame with the designed embellishments and the app would just render inside a rectangle. I talked to @david-gauquelin about this and he plans to document it as part of the design system. Then we can refactor the code for the objects to work that way.

michelleplur commented:
Looks really good ❤️

david-gauquelin commented:
ok so answering all the different points in the thread here:

- yes I'm defining in the guidelines what belongs to the chat and what belongs to a Waku Object (in chat bubble), but definitely the time stamp belongs to the chat app so it should appear on Waku Object messages in-chat
- the floating day/date (always visible) would be really nice as otherwise I just see the time and it can be confusing/misleading
- the time on one line is a nice to have, having on 2 lines is less space-efficient but doesn't create usability issues
- finally, please use a sans-serif font for the timestamp, the serif font is for the message text only
